### PR TITLE
Add sync method

### LINKED ISSUE #3: add oslocale.sync()
For synchronous tools like yargs, it would be awesome to have a synchronous mode for locale lookups.

### PR DESCRIPTION
**Pull request by kevva**
Might be possible to extract out some common code but couldn't find any obvious stuff.

Fixes #3.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,3 +65,42 @@ module.exports = function (cb) {
...
+
+	var env = process.env;
+	var locale = env.LC_ALL || env.LANGUAGE || env.LANG || env.LC_MESSAGES;
+
+	if (locale) {
+		cache = getLocale(locale);
+		return cache;
+	}
+
+	var fn = childProcess.execFileSync || execFileSync;
```
Comment (sindresorhus) [inline index.js]: You don't have to do that. Handled by the ponyfill.

File: index.js [hunk 2]
```diff
@@ -64,3 +65,42 @@ module.exports = function (cb) {
...
+		return cache;
+	};
+
+	if (process.platform === 'win32') {
+		var lcidCode = parseInt(stdout.replace('Locale', ''), 16);
+		cache = lcid.from(lcidCode) || 'en_US';
+		return cache;
+	}
+
+	var res = /(?:LC_ALL|LANG|LC_MESSAGES|LC_CTYPE|)="([^"]{2,})"/.exec(stdout);
```
Comment (sindresorhus) [inline index.js]: This is common between them.

File: index.js [hunk 3]
```diff
@@ -64,3 +65,42 @@ module.exports = function (cb) {
...
 	}
 };
+
+module.exports.sync = function () {
+	if (cache) {
+		return cache;
+	}
+
+	var env = process.env;
+	var locale = env.LC_ALL || env.LANGUAGE || env.LANG || env.LC_MESSAGES;
```
Comment (sindresorhus) [inline index.js]: This is common between them.

File: index.js (hunk 2 above)
Comment (sindresorhus) [inline index.js]: Maybe have a shared processing function that both async and sync uses.

File: index.js (hunk 1 above)
Comment (kevva) [inline index.js]: Oh, I thought I checked, but apparently I'm blind :). I see it now.

File: index.js (hunk 2 above)
Comment (kevva) [inline index.js]: Hmm, it feels like that would get ugly, no?

**Comment by sindresorhus** [comment]
@kevva Thanks!

This is what I meant with DRYing it up: https://github.com/sindresorhus/os-locale/commit/cf130b6e79b0daf5b38092eda0fcca91a48340ed Looks good to you?

File: index.js [hunk 4]
```diff
@@ -64,3 +65,41 @@ module.exports = function (cb) {
...
+		cache = getLocale(locale);
+		return cache;
+	}
+
+	var stdout = process.platform === 'win32' ?
+		execFileSync('wmic', ['os', 'get', 'locale'], {encoding: 'utf8'}) :
+		execFileSync('locale', {encoding: 'utf8'});
+
+	var getAppleLocale = function () {
+		cache = execFileSync('defaults', ['read', '-g', 'AppleLocale']).strim() || 'en_US';
```
Comment (sindresorhus) [inline index.js]: `strim()` => `trim()` and missing encoding option. Fixed in https://github.com/sindresorhus/os-locale/commit/cf130b6e79b0daf5b38092eda0fcca91a48340ed

**Comment by kevva** [comment]
Looks good yo. Thought you meant having a shared function for the other stuff too.

**Pull request merged**